app/models/asset.rb

```diff
@@ -52,6 +52,14 @@ class Asset
     event :scanned_infected do
       transition any => :infected
     end
+
+    event :upload_success do
+      transition %i(clean uploaded not_uploaded) => :uploaded
+    end
+
+    event :upload_failure do
+      transition %i(clean uploaded not_uploaded) => :not_uploaded
```

Comment:
Similar to the question above, do we definitely need to be able to transition from `uploaded` to `not_uploaded` and from `not_uploaded` to `not_uploaded`?

Reply:
`uploaded` -> `not_uploaded`: If we had a Rake task that was re-uploading the existing files and the new upload failed, we might need this.

`not_uploaded` -> `not_uploaded`: If we had a Rake task that was re-uploading the existing files for assets which had previously failed to upload (and none of the Sidekiq retrying had worked) and the new upload failed, we might need this.

What are the benefits of preventing the transitions you mention?